### ISSUE TITLE
Got mercurial hook working, with improvements - [closed]

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#835]
In GitLab by @dhandy on Dec 17, 2016, 18:00

_Merges bug/283 -> master_

This commit fixes a problem that caused "flake8 --install-hook mercurial" to
die with this message:

    Could not find the mercurial directory

Also fixes two usages of configparser (can't supply a non-text value when
calling .set(), and need to use .getboolean() to convert the value to a
boolean.)

This commit converts the hook itself from a function hook to a script hook.
This enables the hook to run even when flake8 is installed in a virtual
environment and Mercurial is installed globally (a common case.) This also
makes it possible for flake8 to be installed for Python 3; Mercurial is
Python 2.x only.

Also convert the hook event from "commit" to "pretxncommit" so that strict
mode would actually work. A regular commit hook runs only after the commit
is completed; a pretxncommit hook will cause the transaction to be rolled
back if it exits with a non-zero code.

Finally, put extensions on the hook names: pretxncommit.flake8 and
qrefresh.flake8. This allows the flake8 hook to coexist with other hooks the
user might have put in place.

I tested flake8 mercurial hook installation and execution on both Python 2.7
and Python 3.5. I did *not* test the qrefresh.flake8 hook. I could not find
documentation on qrefresh hook event; I presume it has something to do with
the Mercurial MQ extension, which I have never used. Probably this commit
does not make it any worse than it was before.

For more info see:
http://hgbook.red-bean.com/read/handling-repository-events-with-hooks.html

Closes #283

**Comment by asottile** [comment]
In GitLab by @dhandy on Dec 17, 2016, 18:37

added 1 commit

* d1b77726 - Got mercurial hook working, with improvements

[Compare with previous version](https://gitlab.com/pycqa/flake8/merge_requests/162/diffs?diff_id=1925947&start_sha=cdd4c6553eb25cc80b6d825581f871530b10cab9)

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Dec 20, 2016, 15:28

So there are some rather drastic changes from the behaviour of this plugin since it was introduced.  I'd rather have a different hook for some of the changes you've made here. If you're not up for it, I'll split it out into another hook.

**Comment by asottile** [comment]
In GitLab by @dhandy on Dec 20, 2016, 19:30

I'm not sure what you mean by "a different hook for some of the changes you've made". Please explain (in words or code changes, whichever is easier.)

**Comment by asottile** [comment]
In GitLab by @asottile on Aug 18, 2019, 15:17

via #568 -- we're deprecating `--install-hook` (in the case of mercurial: without replacement) as it was never well loved and at least for `git` there are better alternatives

thanks for the patch nonetheless!

**Comment by asottile** [comment]
In GitLab by @asottile on Aug 18, 2019, 15:17

closed
